pkg/controller/deployment/credentials_request.go

```diff
@@ -0,0 +1,105 @@
+
+	v1 "github.com/openshift/api/operator/v1"
+)
+
+const (
+	// credentials for AWS
+	awsCredentialsDir = "/.aws"
+
+	// credentials for GCP
+	gcpCredentialsDir       = "/.config/gcloud"
+	gcpCredentialsFileName  = "application_default_credentials.json"
+	gcpCredentialsSecretKey = "service_account.json"
+
+	// cloudCredentialsVolumeName is the volume name for mounting
+	// service account (gcp) or credentials (aws) file
+	cloudCredentialsVolumeName = "cloud-credentials"
+)
+
+func withCloudCredentials(secretsInformer coreinformersv1.SecretInformer, infraInformer configinformersv1.InfrastructureInformer, deploymentName, secretName string) func(operatorSpec *v1.OperatorSpec, deployment *appsv1.Deployment) error {
+	// cloud credentials is only required for the controller deployment,
+	// other deployments should be left untouched
+	if deploymentName != "cert-manager" {
+		return func(operatorSpec *v1.OperatorSpec, deployment *appsv1.Deployment) error {
+			return nil
+		}
+	}
+
+	return func(operatorSpec *v1.OperatorSpec, deployment *appsv1.Deployment) error {
+		if len(secretName) == 0 {
+			return nil
+		}
+
+		_, err := secretsInformer.Lister().Secrets(operatorclient.TargetNamespace).Get(secretName)
+		if err != nil && apierrors.IsNotFound(err) {
+			return fmt.Errorf("(Retrying) cloud secret %q doesn't exist due to %v", secretName, err)
+		} else if err != nil {
+			return err
+		}
+
+		infra, err := infraInformer.Lister().Get("cluster")
+		if err != nil {
+			return err
+		}
+
+		switch infra.Status.PlatformStatus.Type {
+		// supported cloud platform for mounting secrets
+		case configv1.AWSPlatformType:
+			deployment.Spec.Template.Spec.Volumes = append(
+				deployment.Spec.Template.Spec.Volumes,
+				corev1.Volume{
+					Name: cloudCredentialsVolumeName,
+					VolumeSource: corev1.VolumeSource{
+						Secret: &corev1.SecretVolumeSource{
+							SecretName: secretName,
+						},
+					},
+				},
+			)
+			deployment.Spec.Template.Spec.Containers[0].VolumeMounts = append(
+				deployment.Spec.Template.Spec.Containers[0].VolumeMounts,
+				corev1.VolumeMount{
+					Name:      cloudCredentialsVolumeName,
+					MountPath: awsCredentialsDir,
+				},
+			)
+		case configv1.GCPPlatformType:
+			deployment.Spec.Template.Spec.Volumes = append(
+				deployment.Spec.Template.Spec.Volumes,
+				corev1.Volume{
+					Name: cloudCredentialsVolumeName,
+					VolumeSource: corev1.VolumeSource{
+						Secret: &corev1.SecretVolumeSource{
+							SecretName: secretName,
+							Items: []corev1.KeyToPath{{
+								Key:  gcpCredentialsSecretKey,
+								Path: gcpCredentialsFileName,
+							}},
+						},
+					},
+				},
+			)
+			deployment.Spec.Template.Spec.Containers[0].VolumeMounts = append(
+				deployment.Spec.Template.Spec.Containers[0].VolumeMounts,
+				corev1.VolumeMount{
+					Name:      cloudCredentialsVolumeName,
+					MountPath: gcpCredentialsDir,
+				},
+			)
+		}
```

Comment:
Can you refactor this to so that we don't have multiple `append()`. Create a ref for volume and volumeMount outside the switch. 

Also return error on `default` in the switch, we need to catch use cases of other platforms that are using credentials (STS). WDYT?

Reply:
Cool, used:
```
var volume *corev1.Volume
var volumeMount *corev1.VolumeMount
```
and a single set of appends towards the end of the func.

Regarding, `default` case updated it return an error, this would also help discourage users not on AWS/GCP from using that env var for mounting secrets, thanks!